main.tf

```diff
@@ -149,7 +150,7 @@ resource "aws_dms_replication_instance" "this" {
 ################################################################################
 
 resource "aws_dms_endpoint" "this" {
-  for_each = { for k, v in var.endpoints : k => v if var.create }
+  for_each = { for k, v in var.endpoints : k => v if var.create && var.create_endpoints }
```

Comment:
if you don't provide any values to `var.endpoints` then endpoints are not created - what value are these additional variables adding here?